File: array/80.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        if not nums:
            return 0
        length = len(nums)
        pre = 0
        cur = 1
        flag = False #False 1连续 True 2连续
        while cur < length:
            if nums[pre] == nums[cur]:
                if flag:
                    nums.pop(cur)
                    length -= 1
                else:
                    flag = True
                    pre += 1
                    cur += 1
            else:
                flag = False
                pre += 1
                cur += 1
                
        return len(nums)
                    
                    
                
            
        

File: array/80.go
func removeDuplicates(nums []int) int {
    if len(nums) == 0 {
        return 0
    }
    cur := nums[0]
    count := 1
    flag := 1
    index := 1
    for index < len(nums) {
        if nums[index] != cur {
            cur = nums[index]
            flag = 1
            count++
            index++
        }else {
            if flag < 2 {
                flag++
                index++
                count++
            }else{
                // 依次摘除多余的 这里index不需要变化
                for index < len(nums) && nums[index] == cur {
                    nums = append(nums[:index], nums[index+1:]...)
                }
            }
        }
    }
    return count
}
